csrank/discretechoice/generalized_nested_logit.py

```diff
@@ -256,30 +248,33 @@ def construct_model(self, X, Y):
             -------
              model : pymc3 Model :class:`pm.Model`
         """
+        self.trace_ = None
+        self.trace_vi_ = None
         self.random_state_ = check_random_state(self.random_state)
         self.loss_function_ = likelihood_dict.get(self.loss_function, None)
-        if np.prod(X.shape) > self.threshold:
-            upper_bound = int(self.threshold / np.prod(X.shape[1:]))
+        self.threshold_ = 43e5
```

Comment:
Peculiar constant - this should rather be something passed into `__init__` as parameter. But this is something for a separate issue.